src/Microsoft.AspNetCore.Mvc.Razor.Host/Internal/RazorCodeDocumentExtensions.cs

```diff
@@ -30,5 +31,25 @@ public static void SetRelativePath(this RazorCodeDocument document, string relat
 
             document.Items[RelativePathKey] = relativePath;
         }
+
+        public static void SetPageDirective(this RazorCodeDocument document, PageDirective directive)
```

Comment:
I know these types got moved, but putting it out there to figure out if this is the right approach.